app/src/main/java/com/example/poste/activities/EditPostActivity.java

```diff
@@ -59,84 +39,73 @@ public class EditPostActivity extends AppCompatActivity {
+            finish();
+            returnToFolderIntent.putExtra("postID",postId);
+            returnToFolderIntent.putExtra("folderID",folderId);
+            startActivity(returnToFolderIntent);
         });
-
-        // Save button click handler
-        saveBtn.setOnClickListener(view -> {
-            try {
-                String selectedFolderName = spinner.getSelectedItem().toString();
-                Integer selectedFolderId = folderIdNameMap.get(selectedFolderName);
-
-                // Save the post in the API
-                if (currentPost.getName() != postNameField.getText().toString() ||
-                        currentPost.getLink() != postLinkField.getText().toString()) {
-                    API.updatePost(currentPost.getId(), postNameField.getText().toString(), postLinkField.getText().toString(), currentPost.getOwnerId());
-                }
-
-                // Update the post in the API
-                if (currentFolder.getId() != selectedFolderId) {
-                    API.removePostFromFolder(currentPost.getId(), currentFolder.getId());
-                    API.addPostToFolder(currentPost.getId(), selectedFolderId);
-                }
-
-                // Send back to folder view
-                sendToDashboard();
-            } catch (APIException e) {
-                throw new RuntimeException(e);
+        buttonSaveChanges.setOnClickListener(view -> {
+            if (currentPost!= null){
+                String title = postTitle.getText().toString();
+                String description = postDescription.getText().toString();
+                String url = linkToPost.getText().toString();
+
+                MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+                Call<ResponseBody> call = apiService.editPost(User.getUser().getTokenHeader(),
+                        postId,
+                        new EditPostRequest(title,description,url));
+
+                call.enqueue(new Callback<ResponseBody>() {
+                    @Override
+                    public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                        if (response.isSuccessful()){
+                            currentPost.setTitle(title);
+                            currentPost.setDescription(description);
+                            currentPost.setUrl(url);
+                            Toast.makeText(EditPostActivity.this,"Edit post successful!", Toast.LENGTH_SHORT).show();
+                        } else {
+                            Toast.makeText(EditPostActivity.this,"Post to edit not found!", Toast.LENGTH_SHORT).show();
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<ResponseBody> call, Throwable t) {
+                        Toast.makeText(EditPostActivity.this, "Edit failed due to server error", Toast.LENGTH_SHORT).show();
```

Comment:
I was reading on this yesterday and the onFailure here is called when the Response itself fails; could be due to connectivity or client-side failure, too, but generally means the connection to the server never happened

Reply:
I've made the message more ambigious since to be more accurate